pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -236,6 +240,10 @@ func (m *manager) Start(activePods ActivePodsFunc, sourcesReady config.SourcesRe
 }
 
 func (m *manager) Allocate(p *v1.Pod, c *v1.Container) error {
+	// The pod during the admission phase, we need to add the pod to admitted pods
+	// to avoid the pod deletion from the state
+	m.updateAdmittedPods(p)
+
```

Comment:
It feels awkward to have this function called in 3 places. It makes sense for `removeStaleState()` because pods could disappear at any time and we want to make sure we remove the stale state before continuing with the logic in any of these functions. But for updating a state variable about which pod is currently being admitted -- it feels weird to have a "catch all" function that is called in multiple places like this.

Reply:
I can refactor `removeStaleState()` easily to pass the pod to it because it is called under the CPU manager reconcile loop. And I need to call pod update because it is possible that the code will call all three methods as a first method.
1. Allocate - No topology manager, CPU manager with the static policy
2. GetTopologyHints - Topology manager with the container scope
3. GetPodTopologyHints - Topology manager with the pod scope

Reply:
I wonder if it makes sense to track this variable in the `TopologyManager` instead and add a function to the `HintProvider` (or even introduce a new interface) to return it for use in these components.

Reply:
But the interface will not call it for you, you still will need to call the interface method. And I think your question is somehow related to the comment https://github.com/kubernetes/kubernetes/pull/103979#discussion_r679909749

Reply:
I’m not sure what you mean about “the interface will not call it for you”. The idea is that the topology manager would track the variable and the interface call would be something like “GetPodBeingAdmitted()” to just grab access to it when needed.

Reply:
I'd rather have a (possibly undesired) API call creep into the internal `HintProvider` interface (that we can refactor later), than have 9 unnecessary calls spread across 3 different components that all do the same thing.

Especially since the change to implement this new call is to add a single state variable to the `TopologyManager` and `FakeTopologyManager` that gets assigned in the `Admit()` call and then extracted via the new API call (by the original three components).

I picture a refactor later that abstracts how we get the list of pods we actually care about tracking state for (i.e. an interface that returns (1) the currently being admitted pod, (2) active pods, and (3) terminated but still running pods). Having (1) in a single place already, will help make this refactoring easier down the road, for just a little bit more effort now.

Reply:
Since we are discussing this topic, I just want to make sure we take the case "topology manager disabled" into account:
https://github.com/kubernetes/kubernetes/blob/v1.22.0-rc.0/pkg/kubelet/cm/container_manager_linux.go#L747
sorry if being redundant, was bite by this case lately and just trying to save us some more pain :)

Reply:
Right ... that adds a complication to what I suggested before (since it's possible that the feature gate is disabled so *neither* the normal topology manager or the fake topology manager is in use).

Reply:
OK. I'll concede not to do this generalization now until we figure out a better abstraction for it.